quickwit-indexing/src/actors/pipeline_supervisor.rs

```diff
@@ -240,7 +331,32 @@ impl IndexingPipelineSupervisor {
             // }
         } else {
             match self.healthcheck() {
-                Health::Healthy => {}
+                Health::Healthy => {
+                    // We stop triggering new merges as soon as there cannot be more splits.
+                    if let Some(handlers) = self.handlers.as_mut() {
+                        // If the uploader has terminated (due to end of source) we want to stop
+                        // merging new splits.
+                        //
+                        // This is done by sending a message to the `MergePlanner`.
+                        // We already send this message in the uploader finalizer, but if the
+                        // uploader panicks, the finalizer may never be
+                        // called, so we defensively send the message if we
+                        // detect that the uploader is not running, while the merge planner is.
+                        if handlers.uploader.health() != Health::Healthy
```

Comment:
check the packager instead? 
I'm not sure what is the best: send the message in packager finalizer or in the uploader finalizer. Doing it as soon as possible sounds good for me, so we can keep it into the packager and update the comments and health check.

Reply:
Great point